.github/workflows/build-design-system.yml

```
@@ -16,19 +16,36 @@ jobs:
+      if: steps.cache-deps.outputs.cache-hit != 'true'
       working-directory: ./packages/design-system
       run: bun install
 
-    # Step 4: Run the build script
+    # Step 5: Run the build script
     - name: Run build script
       working-directory: ./packages/design-system
       run: bun run build
+
+    # Step 6: Check for changes
+    - name: Check for changes 
```

actionlint (1.7.4)

step must run script with "run" section or run action with "uses" section

(syntax-check)